### PR TITLE
NF: Adding correct_mask to median_otsu

### PR DESCRIPTION
**Pull request by pjsjongsung**
This PR aims to add a function that removes holes or islands of a binary mask to median_otsu.

The function was already being used in
dipy/nn/utils.py, so it has been moved to dipy/segment/utils.py, as it makes more sense for mask correction to be in segmentation module.

Slice-wise hole removal was excluded from the function, since it can cause huge false positives if there is a lack of axial slices.

Finally, the function's name has been changed to
remove_holes_and_islands, to be more explicit about the functionality.

This PR is related to #3124

### CODE REVIEW

**Comment by arokem** [comment]
Could you please add a test of `remove_holes_and_islands`?

File: dipy/nn/evac.py [hunk 1]
```diff
@@ -297,7 +298,7 @@ def __predict(self, x_test):
     def predict(self, T1, affine,
                 voxsize=(1, 1, 1), batch_size=None,
                 return_affine=False, return_prob=False,
-                largest_area=True):
```
Comment (skoudoro) [inline dipy/nn/evac.py:300]: You might have to use `deprecate_params` since this method is available the 2 last release at least.

Comment (skoudoro) [reply dipy/nn/evac.py:300]: to discuss

File: dipy/segment/mask.py [hunk 2]
```diff
@@ -151,9 +152,12 @@ def median_otsu(input_volume, vol_idx=None, median_radius=4, numpass=4,
         if True, the masked input_volume will also be cropped using the
         bounding box defined by the masked data. Should be on if DWI is
         upsampled to 1x1x1 resolution. (default: False).
-
     dilate : None or int, optional
         number of iterations for binary dilation
+    correct_mask : bool, optional
+        Whether to remove potential holes or islands.
+        Useful for solving minor errors.
+        Default is True
```
Comment (skoudoro) [inline dipy/segment/mask.py:160]: default to remove

**Review by skoudoro — COMMENTED** [review]
Hi @pjsjongsung,

Thank you for this. See below for minor comments

File: dipy/nn/evac.py (hunk 1 above)
Comment (skoudoro) [reply dipy/nn/evac.py:300]: here it is : https://github.com/dipy/dipy/blob/90489e87b5496e050d03b1b53be2ca4892982ff0/dipy/reconst/forecast.py#L49

**Comment by skoudoro** [comment]
Can you also rebase or merge master?  Thank you

File: dipy/segment/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import warnings
```
Comment (skoudoro) [inline dipy/segment/tests/test_utils.py:1]: you need to add this file in its own `meson.build` too

**Review by skoudoro — COMMENTED** [review]
Hi @pjsjongsung,

Sorry for the late review. See below some comments. It is almost ready to go

**Comment by skoudoro** [comment]
Hi @pjsjongsung,

I did not look yet but your test is failing:

```python
=========================== short test summary info ============================
FAILED segment/tests/test_utils.py::test_remove_holes_and_islands - AssertionError: 
Arrays are not equal

Mismatched elements: 34970 / 64000 (54.6%)
Max absolute difference: 1.0
Max relative difference: 1.0
 x: array([[[1, 1, 1, ..., 0, 1, 1],
        [1, 1, 1, ..., 0, 1, 1],
        [1, 1, 1, ..., 1, 0, 1],...
 y: array([[[ 0.,  0.,  0., ...,  0.,  0.,  0.],
        [ 0.,  0.,  0., ...,  0.,  0.,  0.],
        [ 0.,  0.,  0., ...,  0.,  0.,  0.],...
===== 1 failed, 1020 passed, 65 skipped, 8 warnings in 1785.02s (0:29:45) ======
```

Can you look closely at this PR? thanks !

**Comment by pjsjongsung** [comment]
My bad. I didn't think the test would fail since it worked before when it was in a different module. Changed the test so that it would not create non realistic tasks where the method would fail.

**Comment by skoudoro** [comment]
Can you rebase and address all the conflict here @pjsjongsung? 

thank you!

**Comment by pjsjongsung** [comment]
It seems there are overlapping changes with #3343 so I'll fix it after that PR is merged

**Comment by skoudoro** [comment]
https://github.com/dipy/dipy/pull/3343 has been merged. Do you have time to finalize this PR?

**Comment by pjsjongsung** [comment]
Instead of moving the function from nn utils, I had to create a different version of the similar function in segment utils since DWI and T1 tends to have different situations where somethings might help and something might not. For clarification, please check the difference between `dipy.nn.utils.correct_minor_errors` and `dipy.nn.segment.remove_holes_and_islands`

**Comment by pjsjongsung** [comment]
It seems there is a conflict between the `@deprecated_params` and keyword only arguments.

../venv/lib/python3.10/site-packages/dipy/nn/evac.py:378
  /Users/runner/work/dipy/dipy/venv/lib/python3.10/site-packages/dipy/nn/evac.py:378: UserWarning: Pass ['new_name'] as keyword args. From version 2.0.0 passing these as positional arguments will result in an error. 
    @deprecated_params("largest_area", "finalize_mask", since="1.10", until="1.12")

It's a warning, but do we want to solve this somehow? @skoudoro

**Review by skoudoro — APPROVED** [review]
LGTM! Thank you for updating this @pjsjongsung, 

merging

**Pull request merged**